src/lib/components/collections/req-res-section/sub-components/close-popup/ClosePopup.svelte

```diff
@@ -20,16 +20,58 @@
     workspaceLevelPermissions,
   } from "$lib/utils/constants/permissions.constant";
   import type { WorkspaceRole } from "$lib/utils/enums";
+  import type { Observable } from "rxjs";
+  import type { CollectionDocument } from "$lib/database/app.database";
+  import { onDestroy } from "svelte";
   export let collectionsMethods: CollectionsMethods;
   export let closeCallback;
   export let componentData: NewTab;
   export let handleSaveAsBackdrop;
   export let onFinish = (_id) => {};
   export let loggedUserRoleInWorkspace: WorkspaceRole;
+  export let _collectionListViewModel;
+
+  const collections: Observable<CollectionDocument[]> =
+    _collectionListViewModel.collection;
+
+  let collectionCountArr = [];
+  let currentCollection;
+  const refreshCount = async () => {
+    if (collectionCountArr && componentData?.path?.collectionId) {
+      for (const collection of collectionCountArr) {
+        if (collection._data.id === componentData?.path?.collectionId) {
+          currentCollection = collection;
+        }
+      }
+    }
+  };
+  const collectionSubscribe = collections.subscribe(
+    (value: CollectionDocument[]) => {
+      if (value) {
+        collectionCountArr = value;
+        refreshCount();
+      }
+    },
+  );
+
+  $: {
+    if (componentData?.path?.collectionId) {
+      refreshCount();
+    }
+  }
 
   let loader: boolean = false;
 
   const handleSaveRequest = async () => {
+    let userSource = {};
+    if (componentData?.activeSync && componentData?.source === "USER") {
```

Comment:
if the `ClosePopup` is a dumb component if it has `activeSync` and `branch` information in state, it doesn't make sense.